### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by Coolnesss** [delta-lang/delta#27]
- [x] Check that the compiler builds on Windows with at least Visual Studio.
- [x] Make the Delta standard library compile on Windows.

**Comment by Coolnesss** [comment]
Tried this with GCC, some things:

* Need LLVM 5.0 to run, which [MinGW](https://en.wikipedia.org/wiki/MinGW) doesn't have in repos
* LLVM 5 binaries can be installed easily but they lack `.cmake` files required for the build. Maybe we can copy these in from elsewhere?
* LLVM can be built from source, which would include these .cmake files, but I couldn't get this to work because of difficulties with dependencies

Otherwise, MinGW can be used to easily install all other dependencies.

**Comment by emillaine** [comment]
I managed to build LLVM and Clang from source on Windows and compile Delta successfully with them. I have only tried with Visual Studio so far but I think it should work with Makefiles or Ninja as well. There are still ~71~ ~58~ ~9~ 2 failing tests, which I'm now trying to fix.

The relevant WIP work is on the [windows](https://github.com/delta-lang/delta/tree/windows) branch. Instructions are in the readme.
